docs/proposals/rescheduling/README.md

```diff
@@ -0,0 +1,220 @@
+---
+title: Rescheduling mechanism based on application health status
+authors:
+  - "@Poor12"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+# Rescheduling mechanism for unhealthy application state
+
+## Summary
+
+In the multi-cluster scenario, user workloads may be deployed in multiple clusters to improve service high availability. Now Karmada already supports multi-cluster failover when detecting a cluster fault.
+However, the application may still be unavailable when the cluster is in a healthy state. For example, the resources of the cluster are reclaimed so that the application has no resources to run.
+Therefore, Karmada needs to provide a means of fault migration from an application perspective.
+
+## Motivation
+
+The cluster administrators want to ensure that the application can run normally, regardless of which cluster the application is running. When the application is unavailable,
+they want Karmada to reschedule the application and avoid clusters that failed to run before.
+
+## Goals
+
+- Extend the API of PropagationPolicy to provide users with the rescheduled configuration.
+- Extend the API of ResourceBinding to record previously evicted clusters.
+- Propose the implementation ideas for involved components, including `karmada-controller-manager`, `karmada-webhook` and `karmada-scheduler`.
+For example, `karmada-controller-manager` need to add a component to observe the health status of the application. `Karmada-scheduler` need to
+add a plugin to filter the evicted cluster.
+
+## Non-Goals
+
+## Proposal
+
+### User Stories (Optional)
+
+#### As a user, I use the cloud vendor's spot instance to deploy my application. As a result, the application may fail due to insufficient resources.
+
+Spot instances are a mode of instances that feature a discounted price and a system interruption mechanism, which means that you can purchase a spot instance ar a discounted price, but the system may automatically repossess it.
```

Comment:
I'm also confused about this story, it seems like descheduler could solve this problem by rescheduling replicas from cluster with insufficient resources. What's the difference?

Reply:
The biggest difference is that the system will allocate resources to you only after you have scheduled it.

Reply:
I understand that this case is dedicated to `serverless cluster` users who bought instances at a limited price, but the cluster will hang the application if other users request for the instance with a higher price.

@Poor12 Could you please add links to some cloud providers' services? like AWS, Google.

Reply:
> a spot instance ar a discounted price

What's the `ar` thing?

Reply:
> I understand that this case is dedicated to `serverless cluster` users who bought instances at a limited price, but the cluster will hang the application if other users request for the instance with a higher price.
> 
> @Poor12 Could you please add links to some cloud providers' services? like AWS, Google.

updated.